test/e2e/framework/log.go

```diff
@@ -118,6 +119,70 @@ func By(text string, callback func(), threshold ...time.Duration) {
 	}
 }
 
+// SucceedEventually is a replacement for
+// gomega.Eventually(callback, intervals...).Should(gomega.Succeed(), extra...).
+//
+// In contrast to gomega, it prints intermediate failures at regular intervals
+// to provide early feedback that a test is stuck. The length of that interval
+// can be given as third parameter after the timeout and poll interval. It
+// defaults to 3 * TestContext.SlowStepThreshold.
+//
+// To get timing information for a long running SucceedEventually, wrap it in
+// By.
+func SucceedEventually(callback func(g gomega.Gomega), extra []interface{}, intervals ...interface{}) {
```

Comment:
I need to point out one potential pitfal (will also add to the comment): the callback must use `g.Expect` instead of `ginkgo.Expect` or our framework helper functions like `framework.ExpectNoError`.

That's because the failure must be reported via the failure handler encapsulated inside the Gomega instance, not the global failure handler.

The alternative is to support a `func() error` callback instead or in addition to `func(g gomega.Gomega)`. If we want to support both, callback would have to be of type `interface{}` with runtime type assertion. Not sure whether generics would help.